packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -384,6 +386,33 @@ function commitHookEffectListMount(tag: number, finishedWork: Fiber) {
   }
 }
 
+function commitSnapshotHookEffectList(finishedWork: Fiber) {
+  const updateQueue: FunctionComponentUpdateQueue | null = (finishedWork.updateQueue: any);
+  const lastEffect = updateQueue !== null ? updateQueue.lastEffect : null;
+  if (lastEffect !== null) {
+    const firstEffect = lastEffect.next;
+    let effect = firstEffect;
+    do {
+      if ((effect.tag & HookSnapshot) === HookSnapshot) {
+        // Run snapshot effect
+        const create = effect.create;
+        // There shouldn't be a return
+        const possibleBadValue = create();
+
+        if (__DEV__) {
+          if (typeof possibleBadValue === 'function') {
+            console.error(
+              "useSnapshotBeforeCommit's function must not return anything. Use an effect " +
+                'for when you want to return a function that handles clean-up.',
```

Comment:
This error messaging seems weird.

> Use effect for when you want to return a function that handles clean-up.

That wouldn't be *the reason* you wanted to use an effect hook. They're for different use cases.

I think this method should be rewritten to be clear that we don't allow any side effects requiring cleanup during this phase.

Reply:
I'm not sure what to word this to. Stating that this cannot have side-effects also kind of feels related to the name of this hook too, as if it can't have side-effects, should it refer to `render` in its name?